packages/build/src/plugins_core/functions/utils.js

```diff
@@ -21,7 +21,10 @@ const getRelativeFunctionMainFiles = async function ({ featureFlags, functionsSr
 
   const zisiFeatureFlags = getZisiFeatureFlags(featureFlags)
   const functions = await listFunctions(functionsSrc, { featureFlags: zisiFeatureFlags })
-  return functions.map(({ mainFile }) => relative(functionsSrc, mainFile))
+  const dedupedFunctions = new Map(functions.map((func) => [func.name, func]))
```

Comment:
Just wondering, any reason you are not going with `Set`? Set ensures no duplicates exist by design :)

Reply:
We might have different functions with the same name. By passing an array of entries to `Map`, we guarantee that the last function with the same name takes precedence, which is the order in which zip-it-and-ship-it outputs the files.

Does that answer the question?

Reply:
Yes, thanks for the explanation :)